src/ActiveLogin.Authentication.BankId.Api/Models/SignRequest.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Runtime.Serialization;
+using System.Text;
+
+namespace ActiveLogin.Authentication.BankId.Api.Models
+{
+    /// <summary>
+    /// Sign request parameters.
+    /// </summary>
+    [DataContract]
+    public class SignRequest
+    {
+        /// <summary></summary>
+        /// <param name="endUserIp">
+        /// The user IP address as seen by RP. IPv4 and IPv6 is allowed.
+        /// Note the importance of using the correct IP address.It must be the IP address representing the user agent (the end user device) as seen by the RP.
+        /// If there is a proxy for inbound traffic, special considerations may need to be taken to get the correct address.
+        ///
+        /// In some use cases the IP address is not available, for instance for voice based services.
+        /// In this case, the internal representation of those systems IP address is ok to use.
+        /// </param>
+        /// <param name="userVisibleData">
+        /// The text to be displayed and signed. The text can be formatted using CR, LF and CRLF for new lines.
+        /// </param>
+        ///
+        public SignRequest(string endUserIp, string userVisibleData)
+            : this(endUserIp, userVisibleData, null, null)
+        {
+        }
+
+        /// <summary></summary>
+        /// <param name="endUserIp">
+        /// The user IP address as seen by RP. IPv4 and IPv6 is allowed.
+        /// Note the importance of using the correct IP address.It must be the IP address representing the user agent (the end user device) as seen by the RP.
+        /// If there is a proxy for inbound traffic, special considerations may need to be taken to get the correct address.
+        ///
+        /// In some use cases the IP address is not available, for instance for voice based services.
+        /// In this case, the internal representation of those systems IP address is ok to use.
+        /// </param>
+        /// <param name="userVisibleData">
+        /// The text to be displayed and signed. The text can be formatted using CR, LF and CRLF for new lines.
+        /// </param>
+        /// <param name="personalIdentityNumber">
+        /// The personal number of the user. 12 digits, century must be included (YYYYMMDDSSSC).
+        /// If the personal number is excluded, the client must be started with the AutoStartToken returned in the response.
+        /// </param>
+        public SignRequest(string endUserIp, string userVisibleData, string personalIdentityNumber)
```

Comment:
Great idea with the byte[] on userNonVisibleData.
I'd suggest removing this overload, as it would hinder us to do an overload that take userNonVisibleData as string if we see a need for that in the future.

Reply:
Modified the extensionmethods as well